### ISSUE TITLE
`AggregationsTest#test_immutable_value_objects` failure with ruby 2.5.0rc1 at 5-1-stable

### ISSUE THREAD

**Issue by yahonda** [rails/rails#31508]
Ruby 2.5 introduces `FrozenError` https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/61131 . 

`AggregationsTest#test_immutable_value_objects` failures are reported at 5-1-stable branch using ruby-head. Somehow this failure does not reproduce at Rails master branch.  Note: I assume Rails 5.1 will support Ruby 2.5.

### Steps to reproduce
```ruby
$ rbenv install 2.5.0-rc1 ; rbenv global 2.5.0-rc1 (or install Ruby 2.5.0 rc1 by any other way, I think).
$ cd rails/activerecord
$ git checkout 5-1-stable
$ bundle install
$ for i in sqlite3 mysql2 postgresql; do ARCONN=$i bundle exec ruby -w -Itest test/cases/aggregations_test.rb -n test_immutable_value_objects -v; done
```

### Expected behavior
It should pass

### Actual behavior
```ruby
$ for i in sqlite3 mysql2 postgresql; do ARCONN=$i bundle exec ruby -w -Itest test/cases/aggregations_test.rb -n test_immutable_value_objects -v; done
/home/yahonda/git/rails/activesupport/lib/active_support/core_ext/hash/slice.rb:21: warning: method redefined; discarding old slice
/home/yahonda/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/arel-8.0.0/lib/arel/visitors/informix.rb:21: warning: assigned but unused variable - froms
Using sqlite3
Run options: -n test_immutable_value_objects -v --seed 64373

# Running:

AggregationsTest#test_immutable_value_objects = 0.02 s = F


Failure:
AggregationsTest#test_immutable_value_objects [test/cases/aggregations_test.rb:28]:
[RuntimeError] exception expected, not
Class: <FrozenError>
Message: <"can't modify frozen #<Class:#<Money:0x00005604ccb6d4c0>>">
---Backtrace---
test/cases/aggregations_test.rb:28:in `block (2 levels) in test_immutable_value_objects'
test/cases/aggregations_test.rb:28:in `instance_eval'
test/cases/aggregations_test.rb:28:in `block in test_immutable_value_objects'
---------------


bin/rails test test/cases/aggregations_test.rb:26


Finished in 0.022074s, 45.3022 runs/s, 45.3022 assertions/s.

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
/home/yahonda/git/rails/activesupport/lib/active_support/core_ext/hash/slice.rb:21: warning: method redefined; discarding old slice
/home/yahonda/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/arel-8.0.0/lib/arel/visitors/informix.rb:21: warning: assigned but unused variable - froms
Using mysql2
Run options: -n test_immutable_value_objects -v --seed 33149

# Running:

AggregationsTest#test_immutable_value_objects = 0.03 s = F


Failure:
AggregationsTest#test_immutable_value_objects [test/cases/aggregations_test.rb:28]:
[RuntimeError] exception expected, not
Class: <FrozenError>
Message: <"can't modify frozen #<Class:#<Money:0x00005560b12146b8>>">
---Backtrace---
test/cases/aggregations_test.rb:28:in `block (2 levels) in test_immutable_value_objects'
test/cases/aggregations_test.rb:28:in `instance_eval'
test/cases/aggregations_test.rb:28:in `block in test_immutable_value_objects'
---------------


bin/rails test test/cases/aggregations_test.rb:26


Finished in 0.031043s, 32.2137 runs/s, 32.2137 assertions/s.

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
/home/yahonda/git/rails/activesupport/lib/active_support/core_ext/hash/slice.rb:21: warning: method redefined; discarding old slice
/home/yahonda/.rbenv/versions/2.5.0-rc1/lib/ruby/gems/2.5.0/gems/arel-8.0.0/lib/arel/visitors/informix.rb:21: warning: assigned but unused variable - froms
Using postgresql
Run options: -n test_immutable_value_objects -v --seed 61770

# Running:

AggregationsTest#test_immutable_value_objects = 0.07 s = F


Failure:
AggregationsTest#test_immutable_value_objects [test/cases/aggregations_test.rb:28]:
[RuntimeError] exception expected, not
Class: <FrozenError>
Message: <"can't modify frozen #<Class:#<Money:0x000055f9d056f688>>">
---Backtrace---
test/cases/aggregations_test.rb:28:in `block (2 levels) in test_immutable_value_objects'
test/cases/aggregations_test.rb:28:in `instance_eval'
test/cases/aggregations_test.rb:28:in `block in test_immutable_value_objects'
---------------


bin/rails test test/cases/aggregations_test.rb:26


Finished in 0.070156s, 14.2539 runs/s, 14.2539 assertions/s.

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
$
```
### System configuration
**Rails version**: 5-1-stable branch whose latest commit is 8598e492719dd456298af621463c9813645f0cc9

**Ruby version**:ruby 2.5.0rc1 (2017-12-14 trunk 61243) [x86_64-linux]

**Comment by yahonda** [comment]
Upgraded to  `minitest (5.10.3)` address these failures. I recall https://github.com/rails/rails/pull/29271 upgrades minitest version to run unit tests in random order, which is available on master branch only, is not and should not be available for 5-1-stable branch.

**Comment by yahonda** [comment]
It may be related to https://github.com/seattlerb/minitest/pull/533

**Comment by yahonda** [comment]
I'm going to open a pull request to specify `FrozenError` in the spec to 5-1-stable and would like to get feedbacks.

**Comment by yahonda** [comment]
Opened #31511

**Comment by yahonda** [comment]
Confirmed this failure reproduces when minitest version is 5.6.1 and does not reproduce with minitest 5.7.0. 

* Reproduce
`minitest (5.6.1)`

* Not reproduce
`minitest (5.7.0)`

**Comment by yahonda** [comment]
#31515 should be correct fixes.

**Comment by yahonda** [comment]
Closed since #31520 has been merged to 5-1-stable.
